librato_python_web/instrumentor/web/django_.py

```diff
@@ -71,91 +72,79 @@ def process_exception(self, request, exception):
             telemetry.count('web.errors')
 
 
-def django_inject_middleware(original_method):
-    @wraps(original_method)
-    def decorator(*args, **keywords):
+def django_inject_middleware(original_method, *args, **keywords):
+    if not hasattr(django_inject_middleware, 'injected_middleware'):
+        # Do this the very first time only
         logger.info('injecting AgentMiddleware into django')
+        django_inject_middleware.injected_middleware = True
```

Comment:
Intended behavior? Could be marking as complete before we know that it worked. On the other hand, that could be a nice behavior if following code has a problem, then setting flag here would avoid repeated failures.

Speaking of which... do we want a try...except block in here to keep the wrapper from causing broader problems.